src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/RainbowArmorTask.java

```diff
@@ -0,0 +1,57 @@
+package io.github.thebusybiscuit.slimefun4.implementation.tasks;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.bukkit.Color;
+import org.bukkit.DyeColor;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.LeatherArmorMeta;
+
+import io.github.thebusybiscuit.cscorelib2.collections.LoopIterator;
+import io.github.thebusybiscuit.slimefun4.api.player.PlayerProfile;
+import io.github.thebusybiscuit.slimefun4.implementation.items.armor.RainbowArmorPiece;
+import io.github.thebusybiscuit.slimefun4.implementation.items.armor.SlimefunArmorPiece;
+
+/**
+ * The {@link RainbowArmorTask} is responsible for handling the change in color of any Rainbow Armor piece.
+ *
+ * @author martinbrom
+ */
+public class RainbowArmorTask extends AbstractArmorTask {
+
+    private static final DyeColor[] DYE_COLORS = { DyeColor.RED, DyeColor.ORANGE, DyeColor.YELLOW, DyeColor.LIME, DyeColor.LIGHT_BLUE, DyeColor.PURPLE, DyeColor.MAGENTA };
```

Comment:
This line is way too long and this does not need to be static.

Reply:
> does not need to be static

What do you mean by that?
Do you think the colors should be configurable or? I might be missing what's wrong with the constant being static.

Reply:
Static variables can cause memory leaks due to the way Bukkit is written.
This is completely harmless for primitives but not so much applies for complex types, like this array.

While, this is not the end of the world and there are many places where this trade off is a-okay...
I think here it serves no real benefit.
There is no concrete reasoning as to why the potential memory leak is outweighed by any benefits here.
And more importantly, this is not exactly future-proof, if we intend to add different kinds of Rainbow armor (perhaps a christmas armor for example), we would need to make this non-constant yet again.

Furthermore, this is a singleton class right now.
There won't be another instance of this in the JVM at runtime, which makes the static-ness of this field not provide any of the memory improvements that a static variable would yield.

All in all, there is really no need for this to be static.

Reply:
You can have multiple constructors though.
The colors could be passed via the constructor and then we could have a default constructor which calls `this(...);`

Reply:
howabout a RainbowArmorPiece#getColors then

Reply:
Check latest commit 😉